src/main/java/UserSession.java

```diff
@@ -0,0 +1,159 @@
+        String action = userInput[0];
+
+        // Run the respective methods based on the action
+        switch(action) {
+        case ACTION_DONE:
+            setTaskAsComplete(userInput[1]);
+            break;
+        case ACTION_LIST:
+            printTaskList();
+            break;
+        case ACTION_TODO:
+            createNewTask(userInput, ACTION_TODO);
+            break;
+        case ACTION_EVENT:
+            createNewTask(userInput, ACTION_EVENT);
+            break;
+        case ACTION_DEADLINE:
+            createNewTask(userInput, ACTION_DEADLINE);
+            break;
+        default:
+            break;
+        }
+    }
+
+    // Sets the particular task as done
+    public static void setTaskAsComplete(String taskNumber) {
+        int index = Integer.parseInt(taskNumber) - 1;
+        tasks[index].markAsDone(true);
+
+        // Inform user of success operation
+        System.out.println("\tNice! I've marked this task as done:");
+        System.out.println("\t" + tasks[index].toString());
+    }
+
+    // Prints the whole list of tasks
+    public static void printTaskList() {
+        System.out.println("\tHere are the tasks in your list:");
+
+        // Print each task based on a specified format
+        int iterations = Task.getNumberOfTasks();
+        for (int i = 0; i < iterations; i++) {
+            System.out.println("\t" + tasks[i].toString());
+        }
+    }
+
+    // Creates a new task based on its type
+    public static void createNewTask(String[] inputSegments, String action) {
+        int new_index = Task.getNumberOfTasks();
+        boolean hasReachedSplit = false;
+        StringBuilder newTask = new StringBuilder();
+        StringBuilder newTaskTimeline = new StringBuilder();
+
+        for (String inputSegment : inputSegments) {
+            if(inputSegment.contains("/")) {
+                hasReachedSplit = true;
+            } else if(!hasReachedSplit && !inputSegment.equals(action)) {
+                newTask.append(WHITESPACE).append(inputSegment);
+            } else if(hasReachedSplit)  {
+                newTaskTimeline.append(WHITESPACE).append(inputSegment);
+            }
+        }
+
+        // Creates a new task type based on the type specified
+        switch (action) {
+        case ACTION_TODO:
+            tasks[new_index] = new Todo(newTask.toString());
+            break;
+        case ACTION_EVENT:
+            tasks[new_index] = new Event(newTask.toString(), newTaskTimeline.toString());
+            break;
+        case ACTION_DEADLINE:
+            tasks[new_index] = new Deadline(newTask.toString(), newTaskTimeline.toString());
+            break;
+        default:
+            break;
+        }
+
+        // Inform user of success operation
+        System.out.println("\t" + "Got it. I've added this task: ");
+        System.out.println("\t\t" + tasks[new_index].toString());
+        System.out.println("\tNow you have "+ (new_index + 1)  + " tasks in the list.");
```

Comment:
Could consider extracting this segment into a single method to inform user of success operation